bolt-modules/out/spec/verbose_spec.rb

```diff
@@ -0,0 +1,194 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'bolt/executor'
+require 'bolt/inventory'
+
+describe 'out::verbose' do
```

Comment:
Because the `out::verbose` function [cannot be called from an apply block](https://github.com/puppetlabs/bolt/pull/2914/files#diff-076cb999ac94782339359502948a12d56dd4c15b7accf327ef95570a8f9f4b42R20-R23), we should test that the function errors from an apply block. You can see an example of testing this behavior [here](https://github.com/puppetlabs/bolt/blob/main/bolt-modules/boltlib/spec/functions/set_resources_spec.rb#L98-L104).

We know if we're in an apply block or not by checking `Puppet[:tasks]`, which is `true` if outside an apply block and `false` if inside an apply block.

Reply:
I added the test and tasks still evaluates to true each time (not getting to pry statement on `verbose.rb:21` + print statement right above it. In `verbose.rb` the `unless Puppet[:tasks]` is the same as in `set_resources.rb` and so is the `let(:tasks_enabled) { false }` in verbose_spec.rb. Thoughts/suggestions?